### ISSUE TITLE
Does not ignore non NPM dependencies

### ISSUE THREAD

**Issue by SCjona** [WarningImHack3r/npm-update-dependencies#130]
### What is the issue?

Some dependencies are distributed outside of NPM. A package.json can have the following entires for example:
```
  "components-care": "github:Samedis-care/react-components#master_dist",
  "xlsx": "https://cdn.sheetjs.com/xlsx-0.20.2/xlsx-0.20.2.tgz"
```

The plugin tries to check those for updates too.

For more information see https://docs.npmjs.com/cli/v6/configuring-npm/package-json#dependencies

### What are the steps to reproduce this issue?

Insert one of the above mentioned dependencies into the package.json and the plugin will ask the NPM registry for updates

### What were you expecting to happen?

The dependencies should just be ignored for now. You could add a feature later to support update checking in git repos (via git tags as explained in the NPM docs link above).

### Paste any relevant logs, error output, etc.

```
2024-09-17 21:21:01,361 [ 640901]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Checking for updates for xlsx with comparator https://cdn.sheetjs.com/xlsx-0.20.2/xlsx-0.20.2.tgz
2024-09-17 21:21:01,361 [ 640901]   WARN - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Comparator https://cdn.sheetjs.com/xlsx-0.20.2/xlsx-0.20.2.tgz is not upgradable, removing cached versions for xlsx
```

```
2024-09-17 20:35:05,686 [2102804]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Checking for updates for components-care with comparator github:Samedis-care/react-components#master_dist
2024-09-17 20:35:05,686 [2102804]   WARN - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Comparator github:Samedis-care/react-components#master_dist is not upgradable, removing cached versions for components-care
```

**Comment by WarningImHack3r** [comment]
Although my check for non-upgradable patterns is a bit weak, I do currently ignore patterns starting with `git` or `http`, containing a `/` or not containing any digit.

It's indeed what's shown in the second line of each log: this pattern is not upgradeable, we skip it (the "removing from cache" part is not always what's happening, it should be "removing from cache if it has been in it at some point" to describe what's actually happening more accurately). These dependencies are **not** fetched in the NPM registry, you can figure this out by grepping the requests as you did in the other issue

TL;DR: I'll take a look at the docs to improve my pattern matching, but my current approach does work in your case, what's the issue?

**Comment by SCjona** [comment]
at least in the case of components-care, you still make a request to NPM registry as shown here https://github.com/WarningImHack3r/npm-update-dependencies/issues/129#issuecomment-2356712257

relevant log:
```
2024-09-17 21:22:31,725 [ 731265]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Deprecation status for package components-care before transformation: 
2024-09-17 21:22:31,725 [ 731265]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Deprecation status for package components-care after transformation: null
2024-09-17 21:22:31,725 [ 731265]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No deprecation status found for package components-care
2024-09-17 21:22:34,818 [ 734358]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Checking for updates for components-care with comparator github:Samedis-care/react-components#master_dist
2024-09-17 21:22:34,818 [ 734358]   WARN - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Comparator github:Samedis-care/react-components#master_dist is not upgradable, removing cached versions for components-care
2024-09-17 21:22:43,607 [ 743147]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package components-care
2024-09-17 21:22:43,607 [ 743147]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package components-care
2024-09-17 21:22:43,607 [ 743147]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing command: "npm v components-care dist.tarball"
2024-09-17 21:22:44,696 [ 744236]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Successfully executed "npm v components-care dist.tarball" with output:
2024-09-17 21:22:44,696 [ 744236]   WARN - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Error while executing "npm v components-care dist.tarball":
npm error 404 Not Found - GET https://registry.npmjs.org/components-care - Not found
npm error 404  'components-care
2024-09-17 21:22:44,696 [ 744236]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package components-care, trying all registries
2024-09-17 21:22:44,696 [ 744236]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package components-care in any registry
2024-09-17 21:22:44,696 [ 744236]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Using default registry for package components-care
2024-09-17 21:22:44,696 [ 744236]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - HTTP GET https://registry.npmjs.org/components-care/latest
2024-09-17 21:22:44,940 [ 744480]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing command: "npm v components-care deprecated --registry=https://registry.npmjs.org"
2024-09-17 21:22:45,552 [ 745092]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Successfully executed "npm v components-care deprecated --registry=https://registry.npmjs.org" with output:
2024-09-17 21:22:45,552 [ 745092]   WARN - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Error while executing "npm v components-care deprecated --registry=https://registry.npmjs.org":
npm error 404 Not Found - GET https://registry.npmjs.org/components-care - Not found
npm error 404  'components-care
2024-09-17 21:22:45,552 [ 745092]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Deprecation status for package components-care before transformation: 
2024-09-17 21:22:45,552 [ 745092]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Deprecation status for package components-care after transformation: null
2024-09-17 21:22:45,552 [ 745092]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No deprecation status found for package components-care
2024-09-17 21:22:48,234 [ 747774]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Checking for updates for components-care with comparator github:Samedis-care/react-components#master_dist
2024-09-17 21:22:48,234 [ 747774]   WARN - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.PackageUpdateChecker - Comparator github:Samedis-care/react-components#master_dist is not upgradable, removing cached versions for components-care
```

**Comment by WarningImHack3r** [comment]
Oh! I indeed don't check "upgradeable" versions for the deprecations’ side, you're right!

**Comment by WarningImHack3r** [comment]
I now have 4 things to fix/improve over an evening and 2 issues from you 😂 thanks a LOT for your help!!

**Comment by SCjona** [comment]
you're welcome :smile: but no one said you have to fix this today
